libs/pika/async_mpi/include/pika/async_mpi/mpi_polling.hpp

```diff
@@ -26,32 +26,33 @@
 #include <utility>
 #include <vector>
 
+//#define DISALLOW_EAGER_POLLING_CHECK 1
+
 namespace pika::mpi::experimental {
 
     enum class stream_type : std::uint32_t
     {
         automatic = 0,
-        send,
-        receive,
-        collective,
-        user,
+        send_1,
+        send_2,
+        receive_1,
```

Comment:
I am using them in testing. Since we are currently using the namespace `pika::mpi::experimental` I feel like it ought to be ok to leave some experimental features in until the API has been battle tested, and final decisions made as to what is good and what isn't.